src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -144,141 +175,172 @@ def extract_values(
+            raise InvalidParams(
+                "Cannot filter count_unique by multiple session.status unless it is in groupBy"
+            )
+
+        super().__init__(name, raw_groupby, status_filter)
+
+    status_to_metric_field = {
+        SessionStatus.HEALTHY: MetricField(None, SessionMetricKey.HEALTHY_USER.value),
+        SessionStatus.ABNORMAL: MetricField(None, SessionMetricKey.ABNORMAL_USER.value),
+        SessionStatus.CRASHED: MetricField(None, SessionMetricKey.CRASHED_USER.value),
+        SessionStatus.ERRORED: MetricField(None, SessionMetricKey.ERRORED_USER.value),
+        None: MetricField(None, SessionMetricKey.ALL_USER.value),
+    }
 
 
 class DurationField(Field):
-    def __init__(self, name: SessionsQueryFunction):
-        self.name = name
+    def __init__(self, name: str, raw_groupby: Sequence[str], status_filter: StatusFilter):
         self.op = name[:3]  # That this works is just a lucky coincidence
+        super().__init__(name, raw_groupby, status_filter)
 
-    def get_session_status(
-        self, metric_field: MetricField, raw_groupby: Sequence[str]
-    ) -> Optional[str]:
+    def _get_session_status(self, metric_field: MetricField) -> Optional[SessionStatus]:
         assert metric_field == MetricField(self.op, SessionMetricKey.DURATION.value)
-        if "session.status" in raw_groupby:
-            return "healthy"
+        if "session.status" in self._raw_groupby:
+            return SessionStatus.HEALTHY
         return None
 
-    def get_metric_fields(self, raw_groupby: Sequence[str]) -> Sequence[MetricField]:
-        return [MetricField(self.op, SessionMetricKey.DURATION.value)]
+    def _get_metric_fields(
+        self, raw_groupby: Sequence[str], status_filter: StatusFilter
+    ) -> Sequence[MetricField]:
+        if status_filter is None or SessionStatus.HEALTHY in status_filter:
+            return [MetricField(self.op, SessionMetricKey.DURATION.value)]
 
-    def normalize(self, value: Union[int, float, None]) -> Union[int, float, None]:
+        return []  # TODO: test if we can handle zero fields
+
+    def normalize(self, value: Scalar) -> Scalar:
         value = finite_or_none(value)
         if value is not None:
             value *= 1000
         return value
 
 
-COLUMN_MAP = {
-    SessionsField.name: SessionsField(),
-    UsersField.name: UsersField(),
-    "avg(session.duration)": DurationField("avg(session.duration)"),
-    "p50(session.duration)": DurationField("p50(session.duration)"),
-    "p75(session.duration)": DurationField("p75(session.duration)"),
-    "p90(session.duration)": DurationField("p90(session.duration)"),
-    "p95(session.duration)": DurationField("p95(session.duration)"),
-    "p99(session.duration)": DurationField("p99(session.duration)"),
-    "max(session.duration)": DurationField("max(session.duration)"),
+FIELD_MAP: Mapping[SessionsQueryFunction, Type[Field]] = {
+    "sum(session)": SessionsField,
+    "count_unique(user)": UsersField,
```

Comment:
I don't understand the reason for these two keys becoming hardcoded strings.

Reply:
`name` is now an argument to `__init__`, so I figured we might as well make these entries consistent with the other entries. I forgot to remove the class members though, will do that now.